pallets/gear-voucher/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Gear Voucher Pallet
+//!
+//! The Gear Voucher Pallet provides functionality for alternative source of funds for
+//! gas and transaction fees payment when sending messages to Contracts in Gear engine.
+//! These funds can only be used for a specific purpose - much like a payment voucher.
+//! Hence the pallet name.
+//!
+//! - [`Config`]
+//! - [`Pallet`]
+//!
+//! ## Overview
+//!
+//! This pallet provides API for contract owners (or any other party for that matter)
+//! to sponsor contract users by allocating some funds in such a way that these funds
+//! can only be spent to pay for gas and transaction fees if a user sends a message
+//! to the contract.
+//!
+//! The Gear Voucher Pallet provides functions for:
+//! - Issuing a voucher for an account by any other account.
+//! - Deriving a unique keyless account id for a pair (user, program) to hold allocated funds.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! * `issue` - Issue an `amount` tokens worth voucher for a `user` to be used to pay fees and gas when sending messages to `program`.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::items_after_test_module)]
```

Comment:
why?

Reply:
Otherwise clippy would complain. This is also the case for some other pallets/crates. I frankly fail to see the pattern: in some cases clippy check goes through, in some cases it does not.